tests/chainer_tests/testing_tests/test_matrix.py

```diff
@@ -0,0 +1,73 @@
+import unittest
+
+import numpy
+
+from chainer import testing
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [
+        numpy.float16, numpy.float32, numpy.float64,
+        numpy.complex64, numpy.complex128,
+    ],
+    'x_s_shapes': [
+        ((2, 2), (2,)),
+        ((2, 3), (2,)),
+        ((3, 2), (2,)),
+        ((2, 3, 4), (2, 3)),
+        ((2, 4, 3), (2, 3)),
+        ((0, 2, 3), (0, 2)),
+        # broadcast
+        ((2, 2), ()),
+        ((2, 3, 4), (2, 1)),
+    ],
+}))
+class TestGenerateMatrix(unittest.TestCase):
+
+    def test_generate_matrix(self):
+        dtype = self.dtype
+        x_shape, s_shape = self.x_s_shapes
+        sv = 0.5 + numpy.random.random(s_shape).astype(dtype().real.dtype)
```

Comment:
```suggestion
        sv = numpy.random.uniform(0.5, 1.5, s_shape).astype(dtype().real.dtype)
```
looks more readable to me and returns consistent dtype if `s_shape == ()`.

Reply:
Makes sense, applied. Thanks!